### PR TITLE
Migrate doku - Plugin configuration

### PR DESCRIPTION
**Pull request by swDennis**

### CODE REVIEW

File: source/developers-guide/plugin-configuration/index.md [hunk 1]
```diff
@@ -20,25 +20,36 @@ the configuration options for plugins. Shopware delivers a big amount of helper
 configuration fields in backend.
 
 ## Creating forms
-To create a simple plugin configuration we can use the function `Form()`
+Backend plugin configuration can be extended by Resources/config.xml file.
```
Comment (mitelg) [inline source/developers-guide/plugin-configuration/index.md]: A plugin configuration can be created with a Resources/config.xml file.

File: source/developers-guide/plugin-configuration/index.md [hunk 2]
```diff
@@ -20,25 +20,36 @@ the configuration options for plugins. Shopware delivers a big amount of helper
...
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="https://raw.githubusercontent.com/shopware/shopware/5.2/engine/Shopware/Components/Plugin/schema/config.xsd">
+    <elements>
+        <element required="true" type="text" scope="locale">
+            <name>simpleTextField</name>
+            <label lang="de">Text</label>
+            <label lang="en">Text</label>
+            <value>Vorauswahl</value>
```
Comment (mitelg) [inline source/developers-guide/plugin-configuration/index.md]: please use an english default

File: source/developers-guide/plugin-configuration/index.md [hunk 3]
```diff
@@ -20,25 +20,36 @@ the configuration options for plugins. Shopware delivers a big amount of helper
...
+            <description>Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut.</description>
```
Comment (mitelg) [inline source/developers-guide/plugin-configuration/index.md]: please also translate the description, to make clear that this is possible

File: source/developers-guide/plugin-configuration/index.md [hunk 4]
```diff
@@ -50,53 +61,35 @@ The options parameter of the `setElement` function allows to set several configu
...
- 
-    $form->setElement('text', 'simpleTextField', 
-        array(
-            'label' => 'Text',
-            'value' => 'Vorauswahl'
-        )
-    );
-}
+```xml
+<value>Vorauswahl</value>
```
Comment (mitelg) [inline source/developers-guide/plugin-configuration/index.md]: english default, see above

File: source/developers-guide/plugin-configuration/index.md [hunk 5]
```diff
@@ -50,53 +61,35 @@ The options parameter of the `setElement` function allows to set several configu
...
 ```
 
 <img src="img/default_value.png" alt="Textfield with default value" class="image-border" />
 
+### Required
+The required parameter specifies whether the configuration item is mandatory or not.
```
Comment (mitelg) [inline source/developers-guide/plugin-configuration/index.md]: actually `required` is an attribute

File: source/developers-guide/plugin-configuration/index.md [hunk 6]
```diff
@@ -105,505 +98,192 @@ You can read more on subshop specific configuration at the end of this article r
...
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="https://raw.githubusercontent.com/shopware/shopware/5.2/engine/Shopware/Components/Plugin/schema/config.xsd">
+    <elements>
+        <element required="true" type="text" scope="locale">
+            <name>simpleTextField</name>
+            <label lang="de">Text</label>
+            <label lang="en">Text</label>
+            <value>Vorauswahl</value>
```
Comment (mitelg) [inline source/developers-guide/plugin-configuration/index.md]: english default value, see above

File: source/developers-guide/plugin-configuration/index.md [hunk 7]
```diff
@@ -105,505 +98,192 @@ You can read more on subshop specific configuration at the end of this article r
...
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="https://raw.githubusercontent.com/shopware/shopware/5.2/engine/Shopware/Components/Plugin/schema/config.xsd">
+    <elements>
+        <element required="true" type="text" scope="shop">
+            <name>simpleTextField</name>
+            <label lang="de">Text</label>
+            <label lang="en">Text</label>
+            <value>Vorauswahl</value>
```
Comment (mitelg) [inline source/developers-guide/plugin-configuration/index.md]: english default value

File: source/developers-guide/plugin-configuration/index.md [hunk 8]
```diff
@@ -105,505 +98,192 @@ You can read more on subshop specific configuration at the end of this article r
...
+    </elements>
+</config>
+```
+
+With the help of the scope parameter we can assign configurations per subshop. Without the scope parameter the configuration is used in all subshops. We´ve mentioned this earlier this article.
+
+```php
 public function onPostDispatch(Enlight_Event_EventArgs $arguments)
 {
     $config = $this->Config();
```
Comment (mitelg) [inline source/developers-guide/plugin-configuration/index.md]: please use the config reader service here

File: source/developers-guide/plugin-configuration/index.md [hunk 9]
```diff
@@ -618,12 +298,12 @@ public function onPostDispatch(Enlight_Event_EventArgs $arguments)
  
     $view->addTemplateDir($this->Path() . 'Views/');
```
Comment (mitelg) [inline source/developers-guide/plugin-configuration/index.md]: wrong views path

File: source/developers-guide/plugin-configuration/index.md [hunk 10]
```diff
@@ -618,12 +298,12 @@ public function onPostDispatch(Enlight_Event_EventArgs $arguments)
...
 }
-```
 
-Within the `postDispatch()` method we create an instance of the configuration. Afterwards we test for the just created __show__ attribute. If
-this attribute is empty the method ends and the plugin will not be executed or shown any further. If __show__ is set the template is loaded and 
-assigned via __smarty__.
 ```
+
+Within the onPostDispatch() method in a subscriber we create an instance of the configuration. Afterwards we test for the just created show attribute. If this attribute is empty the method ends and the plugin will not be executed or shown any further. If show is set the template is loaded and assigned via smarty.
```
Comment (mitelg) [inline source/developers-guide/plugin-configuration/index.md]: the `show` attribute is nowhere created in the article

File: source/developers-guide/plugin-configuration/index.md [hunk 11]
```diff
@@ -632,7 +312,6 @@ assigned via __smarty__.
 {/block}
 ```
 
-Our new template extends the parent __logo-container.tpl__ and overwrites the __frontend_index_logo__ block and sets the text we used in the `bootstrap.php`. Now we can open the plugin configuration via
-Plugin Manager and configure it differently for every subshop.
+Our new template extends the parent logo-container.tpl and overwrites the frontend_index_logo block and sets the text we used in the bootstrap.php. Now we can open the plugin configuration via Plugin Manager and configure it differently for every subshop.
```
Comment (mitelg) [inline source/developers-guide/plugin-configuration/index.md]: bootstrap.php?

File: source/developers-guide/plugin-configuration/index.md (hunk 1 above)
Comment (swDennis) [reply source/developers-guide/plugin-configuration/index.md]: d

File: source/developers-guide/plugin-configuration/index.md (hunk 5 above)
Comment (swDennis) [reply source/developers-guide/plugin-configuration/index.md]: d

File: source/developers-guide/plugin-configuration/index.md (hunk 9 above)
Comment (swDennis) [reply source/developers-guide/plugin-configuration/index.md]: d

File: source/developers-guide/plugin-configuration/index.md (hunk 11 above)
Comment (swDennis) [reply source/developers-guide/plugin-configuration/index.md]: D

File: source/developers-guide/plugin-configuration/index.md (hunk 7 above)
Comment (swDennis) [reply source/developers-guide/plugin-configuration/index.md]: d

File: source/developers-guide/plugin-configuration/index.md (hunk 6 above)
Comment (swDennis) [reply source/developers-guide/plugin-configuration/index.md]: d

File: source/developers-guide/plugin-configuration/index.md (hunk 4 above)
Comment (swDennis) [reply source/developers-guide/plugin-configuration/index.md]: d

File: source/developers-guide/plugin-configuration/index.md (hunk 3 above)
Comment (swDennis) [reply source/developers-guide/plugin-configuration/index.md]: d

File: source/developers-guide/plugin-configuration/index.md (hunk 2 above)
Comment (swDennis) [reply source/developers-guide/plugin-configuration/index.md]: d

File: source/developers-guide/plugin-configuration/index.md (hunk 10 above)
Comment (swDennis) [reply source/developers-guide/plugin-configuration/index.md]: d

**Pull request merged**